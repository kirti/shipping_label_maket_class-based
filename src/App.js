import React from 'react'
import Home from './common/Home'

function App() {
	return (
		<>
			<Home />
		</>
	)
}

export default App
