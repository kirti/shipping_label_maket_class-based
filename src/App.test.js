import React from 'react'
import App from './App'
import { render , getByText ,queryByAttribute , fireEvent , cleanup  } from '@testing-library/react';


afterEach(cleanup)

it('renders Login Page', () => {
	const { getByText } = render(<App />)
	const linkElement = getByText(/Shipping Label Maker/i)
	expect(linkElement).toBeInTheDocument()
})

it('Render Sender Component when correct user and pwd  ', () => {
	const { container } = render(<App />);
	const getById = queryByAttribute.bind(null, 'id');
	const username = getById(container, "username");
	const password = getById(container, "password");
	const submitButton = getById(container, "login");
	expect(username.textContent).toBe("");
	expect(password.textContent).toBe("");

	const newName = "kirti";
	fireEvent.change(username, { target: { value: newName } });
	const newPwd = "shiva@3@456";
	fireEvent.change(password, { target: { value: newPwd } });
	fireEvent.click(submitButton)
	const { getByText } = render(<App />)
	const linkElement = getByText(/Enter the sender address/i)
	expect(linkElement).toBeInTheDocument()
  });


  it('Render Login page when incorrect user and pwd  ', () => {
	const { container } = render(<App />);
	const getById = queryByAttribute.bind(null, 'id');
	const username = getById(container, "username");
	const password = getById(container, "password");
	const submitButton = getById(container, "login");
	expect(username.textContent).toBe("");
	expect(password.textContent).toBe("");

	const newName = "kirti2";
	fireEvent.change(username, { target: { value: newName } });
	const newPwd = "shiva@3@456";
	fireEvent.change(password, { target: { value: newPwd } });
	fireEvent.click(submitButton)
	const { asFragment } = render(<App />)
	expect(asFragment(<App />)).toMatchSnapshot()
  });


  it('Render Complete App with all screen with Prev and Next ', async () => {

	const { container  } = render(<App />);
	const getById = queryByAttribute.bind(null, 'id');
	const username = getById(container, "username");
	const password = getById(container, "password");
	const submitButton = getById(container, "login");
	expect(username.textContent).toBe("");
	expect(password.textContent).toBe("");

	const newName = "kirti";
	fireEvent.change(username, { target: { value: newName } });
	const newPwd = "shiva@3@456";
	fireEvent.change(password, { target: { value: newPwd } });
	fireEvent.click(submitButton)
	const { getByText } = render(<App />)
	const linkElement = getByText(/Enter the sender address/i)
	expect(linkElement).toBeInTheDocument()


	const name = getById(container, "name");
	const street = getById(container, "street");
	const city = getById(container, "city");
	const state = getById(container, "state");
	const zip = getById(container, "zip");

	const nextButton = getById(container, "next");
	const prevButton = getById(container, "back");
	

	expect(name.textContent).toBe("");
	expect(street.textContent).toBe("");
	expect(city.textContent).toBe("");
	expect(state.textContent).toBe("");
	expect(zip.textContent).toBe("");
	expect(prevButton).toBeDisabled()

	const newSenderName = "kirti";
	fireEvent.change(name, { target: { value: newSenderName } });
	const newStreet = "toronto";
	fireEvent.change(street, { target: { value: newStreet } });
	fireEvent.click(nextButton)




	const linkElement2 = await getByText(/Enter Receiver's Address gghghghg/i)
	expect(linkElement2).toBeInTheDocument()
	expect(prevButton).not.toBeDisabled()

	const nameReceiver = getById(container, "name");
	const streetReceiver = getById(container, "street");
	const cityReceiver = getById(container, "city");
	const stateReceiver = getById(container, "state");
	const zipReceiver = getById(container, "zip");

	expect(nameReceiver.textContent).toBe("");
	expect(streetReceiver.textContent).toBe("");
	expect(cityReceiver.textContent).toBe("");
	expect(stateReceiver.textContent).toBe("");
	expect(zipReceiver.textContent).toBe("");

	const newNameReceiver = "shiva";
	fireEvent.change(nameReceiver, { target: { value: newNameReceiver } });


	fireEvent.click(nextButton)
	
	const weight = await getById(container, "weight");
	expect(weight.textContent).toBe("");
	const newWeight = "10";
	fireEvent.change(weight, { target: { value: newWeight } });

	const { getByText : weightScreen } = await render(<App />)
	const linkElementR = weightScreen(/Enter Weight fdfgg/i)
	expect(linkElementR).toBeInTheDocument()

	fireEvent.click(nextButton)

	
	const { getByLabelText } = await render(<App />);
	const shippingOption =  getByLabelText('Ground')
	fireEvent.click(shippingOption, { target: { value: "1" } });
	expect(shippingOption.value).toBe('1')
	

	const { getByText : shippingScreen } = await render(<App />)
	const linkElementW = shippingScreen(/Select Shipping Option/i)
	expect(linkElementW).toBeInTheDocument()

	fireEvent.click(nextButton)

 // Checking previous button from step4 
	fireEvent.click(prevButton)
	const { getByText : ShippingScreenExit } = await render(<App />)
	const confirmScreenExist = ShippingScreenExit(/Select Shipping Option/i)
	expect(confirmScreenExist).toBeInTheDocument()


	fireEvent.click(nextButton)
	const { getByText : confirmScreen } = await render(<App />)
	const linkElementConfirm = confirmScreen(/Sender Information/i)
	expect(linkElementConfirm).toBeInTheDocument()

	fireEvent.click(nextButton)
	const { getByText : finalScreen } = await render(<App />)
	const finalScreenElement = finalScreen(/unknown/i)
	expect(finalScreenElement).toBeInTheDocument()
  });


