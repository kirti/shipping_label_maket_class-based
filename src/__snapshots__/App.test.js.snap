// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login page when incorrect user and pwd   1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-55"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorDefault makeStyles-appBar-56 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit"
        >
          Shipping Label Maker
        </h6>
      </div>
    </header>
    <main
      class="MuiGrid-root makeStyles-root-55 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-image-62 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
      />
      <div
        class="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
      >
        <div
          class="makeStyles-paper-58"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-63 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
          <h1
            class="MuiTypography-root MuiTypography-h5"
          >
            Sign in
          </h1>
          <h1
            class="MuiTypography-root MuiTypography-h5"
          />
          <form
            class="makeStyles-form-64"
            novalidate=""
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
                data-shrink="true"
                for="username"
                id="username-label"
              >
                Username
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="username"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-70 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-72 PrivateNotchedOutline-legendNotched-73"
                  >
                    <span>
                      Username *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-70 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-72"
                  >
                    <span>
                      Password *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-74 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-77"
                    data-indeterminate="false"
                    type="checkbox"
                    value="remember"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Remember me
              </span>
            </label>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-65 MuiButton-containedPrimary MuiButton-fullWidth"
              id="login"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                  href="#"
                >
                  Don't have an account? Sign Up
                </a>
              </div>
            </div>
          </form>
        </div>
      </div>
    </main>
    <div>
      <footer
        class="makeStyles-footer-66"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Shipping Label Marker
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            Copyright © 
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="https://material-ui.com/"
            >
              Your Website
            </a>
             2020.
          </p>
        </div>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
